addons/account_payment/models/account_payment.py

```diff
@@ -111,7 +111,7 @@ def _onchange_set_payment_token_id(self):
             self.payment_token_id = False
             return
 
-        self.payment_token_id = self.env['payment.token'].search([
+        self.payment_token_id = self.env['payment.token'].sudo().search([
```

Comment:
I recommend always adding a comment to justify the use of `sudo()`; it makes further maintenance and debugging way easier.
```py
self.payment_token_id = self.env['payment.token'].sudo().search([
            *self.env['payment.token']._check_company_domain(self.company_id),
            ('partner_id', '=', self.partner_id.id),
            ('provider_id.capture_manually', '=', False),
            ('provider_id', '=', self.payment_method_line_id.payment_provider_id.id),
         ], limit=1)  # In sudo mode to read the provider fields.
```